sympy/core/numbers.py

```diff
@@ -1308,73 +1329,98 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self == other
 
-    def __gt__(self, other):
+    def _Frel(self, other, op):
+        from sympy.core.evalf import evalf
+        from sympy.core.numbers import prec_to_dps
         try:
             other = _sympify(other)
         except SympifyError:
             raise TypeError("Invalid comparison %s > %s" % (self, other))
         if other.is_NumberSymbol:
-            return other.__lt__(self)
-        if other.is_Rational and not other.is_Integer:
-            self *= other.q
-            other = _sympify(other.p)
-        elif other.is_comparable:
-            other = other.evalf()
-        if other.is_Number and other is not S.NaN:
+            return other
+        elif other.is_Rational:
+            from sympy.functions.elementary.complexes import sign
```

Comment:
On my system, comparisons like `a < b` (with `a = Float('2/3'), b = Rational(2, 3)`) take over 8 ms. (Probably because logarithms are relatively slow.)
On the other hand, this code

        if other.is_Rational:
            smpf = mlib.mpf_mul(self._mpf_, mlib.from_int(other.q))
            ompf = mlib.from_int(other.p)
            return _sympify(bool(op(smpf, ompf)))

takes only 21 μs.

Reply:
I'm not sure why I saw the difference in run times. I implemented your suggestion and put an inline justifying why we are working with mpf multiplication instead of naive `self*other.q`.